### PR TITLE
Updated the code, now it freezes

### PR DESCRIPTION
**Pull request by alexandruGheorghiu94**
I took the last version of the code and I did the following:
- created separate enums for each register of peripherals in system.h : ePeripheralsRCC_AHBENR, ePeripheralsRCC_APB1ENR & ePeripheralsRCC_APB2ENR
- made individual functions to enable the peripheral in system.c : enablePeripheralRCC_APB1ENR, enablePeripheralRCC_APB2ENR & enablePeripheralRCC_AHBENR

But there is one small problem: the board freezes, the display doesn't work and the RGB LED doesn't light up.

### CODE REVIEW

File: all/projectAlcazar/src/app/project.c [hunk 1]
```diff
@@ -46,6 +51,23 @@
...
 #define UART2_RX 3
 
+//HOMEWORK: functia pentru pwm-ul led-urilor de pe placa
+void initBoardPWM(void)
+{
+	//dam enable la portul c de gpio
+	enablePeripheralRCC_AHBENR(ePerif_GPIOC, eEnabled);
+
+	//setam alternate function pentru PC8 + PC9
+	setPinAlternateFunction(GPIOC, LED_PIN_BLUE, ePin_AF2);
```
Comment (ADiea) [inline all/projectAlcazar/src/app/project.c]: PC8 and PC8 only have ONE alternate function see table 13 in datasheet section 4. So here should be AF0. Only ports A and B have more alternate functions per poin that's why there are 2 tables alternate functions for port A and port B to show which alternate fucntion  is for each value

File: all/projectAlcazar/src/sys/timer.c [hunk 2]
```diff
@@ -51,6 +51,107 @@ unsigned int getTimerValue(volatile Timer_Type* timer)
...
+	//the same expression and shift it by 8 to the left for channel 2
+	//so that we end up modifying bits 15 - 8
+
+	// (4) set the capture and compare mode registers for our two channels
+	timer->CCMR1 =  (TIM_MODE_PWM1 << TIM_OCxM | 1 << TIM_OCxPE) << 8 |  //configuration for channel 4
+					(TIM_MODE_PWM1 << TIM_OCxM | 1 << TIM_OCxPE);      //configuration for channel 3
+
+
+	// (5) enable three channels with normal polarity. see 17.4.9
+	timer->CCER = 1<<TIM_CC1E | 1<<TIM_CC2E;
```
Comment (ADiea) [inline all/projectAlcazar/src/sys/timer.c]: You should use TIM_CC3E and TIM_CC4E which are bits 8 & 12 define them in header file if not defined

File: all/projectAlcazar/src/app/project.c [hunk 3]
```diff
@@ -35,6 +36,10 @@
 #define LED_RGB_GREEN_CH 1
 #define LED_RGB_BLUE_CH 0
 
+//HOMEWORK: PC8 + PC9 sunt definiti pe cananlul TIM3 (n-1 in functie de canal)
+#define LED_PIN_BLUE_CH 2
```
Comment (ADiea) [inline all/projectAlcazar/src/app/project.c]: From the datasheet section 4 table 13
```
PC8 I/O FT - TIM3_CH3 -
PC9 I/O FT - TIM3_CH4
```
So channels are actually 3 and 4

File: all/projectAlcazar/src/sys/system.c [hunk 4]
```diff
@@ -18,28 +18,70 @@ void enablePeripheral(ePeripherals peripheral, eEnableState enable)
 	{
 		pRCCRegister = &RCC_AHBENR;
```
Comment (ADiea) [inline all/projectAlcazar/src/sys/system.c]: no need for if, register will always be RCC_AHBENR

File: all/projectAlcazar/src/sys/system.c [hunk 5]
```diff
@@ -18,28 +18,70 @@ void enablePeripheral(ePeripherals peripheral, eEnableState enable)
...
+		*pRCCRegister &= ~(1 << peripheral);
+	}
+}
+
+
+void enablePeripheralRCC_APB1ENR(ePeripheralsRCC_APB1ENR peripheral, eEnableState enable)
+{
+	volatile unsigned int *pRCCRegister2;
+
+	if(ePerif_USART2EN == peripheral)
```
Comment (ADiea) [inline all/projectAlcazar/src/sys/system.c]: no need for ifs, register will always be RCC_APB1ENR

File: all/projectAlcazar/src/sys/system.c [hunk 6]
```diff
@@ -18,28 +18,70 @@ void enablePeripheral(ePeripherals peripheral, eEnableState enable)
...
+		*pRCCRegister2 &= ~(1 << peripheral);
+	}
+}
+
+
+void enablePeripheralRCC_APB2ENR(ePeripheralsRCC_APB2ENR peripheral, eEnableState enable)
+{
+	volatile unsigned int *pRCCRegister3;
+
+	if(ePerif_TIM1 == peripheral)
```
Comment (ADiea) [inline all/projectAlcazar/src/sys/system.c]: No need for if comparisons register will always be RCC_APB2ENR

File: all/projectAlcazar/src/sys/system.c [hunk 7]
```diff
@@ -18,28 +18,70 @@ void enablePeripheral(ePeripherals peripheral, eEnableState enable)
...
+	{
+		pRCCRegister3 = &RCC_APB1ENR;
```
Comment (ADiea) [inline all/projectAlcazar/src/sys/system.c]: register should be RCC_APB2ENR not RCC_APB1ENR

File: all/projectAlcazar/src/sys/timer.c [hunk 8]
```diff
@@ -51,6 +51,107 @@ unsigned int getTimerValue(volatile Timer_Type* timer)
...
+	 	 	 	 	 	 	 	 as long as TIMx_CNT<TIMx_CCR1 else inactive (LOW or 0).
+	 Bit  7		OCxCE	=> 0: OC1Ref is not affected by the ETRF Input
+	 */
+
+	//we have the same configuration for all 3 channels so we can use
+	//the same expression and shift it by 8 to the left for channel 2
+	//so that we end up modifying bits 15 - 8
+
+	// (4) set the capture and compare mode registers for our two channels
+	timer->CCMR1 =  (TIM_MODE_PWM1 << TIM_OCxM | 1 << TIM_OCxPE) << 8 |  //configuration for channel 4
```
Comment (ADiea) [inline all/projectAlcazar/src/sys/timer.c]: CCMR2 should be used that controls channels 3 & 4 CCMR1 controls channels 1 and 2

File: all/projectAlcazar/src/sys/system.c (hunk 7 above)
Comment (alexandruGheorghiu94) [reply all/projectAlcazar/src/sys/system.c]: This fixed the freezing.

File: all/projectAlcazar/src/app/project.c (hunk 3 above)
Comment (alexandruGheorghiu94) [reply all/projectAlcazar/src/app/project.c]: shouldn't they be channel 2 & 3 ? based on the rgb LED the channels are 0,1,2 for timer1 channels 1,2 & 3. in Timer.c the switch case starts with 0 . 
switch(channel)
	{
		case 0:
			pDutyRegister = & (timer->CCR1);
			break;

**Pull request closed without merging**